mllib/src/test/scala/org/apache/spark/ml/tree/impl/RandomForestSuite.scala

```diff
@@ -176,6 +203,18 @@ class RandomForestSuite extends SparkFunSuite with MLlibTestSparkContext {
     assert(tree.rootNode.impurity === -1.0)
```

Comment:
We should change this test. The reason it didn't fail before is because one of the categorical features ends up being treated as an ordered categorical feature, which always generates splits. You can do that by changing the number of categories to be 1 for the categorical features.